fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java

```diff
@@ -1829,9 +1829,17 @@ public PlanFragment visitPhysicalNestedLoopJoin(
     public PlanFragment visitPhysicalLimit(PhysicalLimit<? extends Plan> physicalLimit, PlanTranslatorContext context) {
         PlanFragment inputFragment = physicalLimit.child(0).accept(this, context);
         PlanNode child = inputFragment.getPlanRoot();
-        child.setLimit(MergeLimits.mergeLimit(physicalLimit.getLimit(), physicalLimit.getOffset(), child.getLimit()));
-        // TODO: plan node don't support limit
-        // child.setOffset(MergeLimits.mergeOffset(physicalLimit.getOffset(), child.getOffset()));
+
+        if (!(inputFragment.getPlanRoot() instanceof ExchangeNode)) {
+            inputFragment.getPlanRoot().setLimit(physicalLimit.getLimit());
+            inputFragment.getPlanRoot().setOffset(physicalLimit.getOffset());
```

Comment:
Here the  inputFragment.getPlanRoot() should replace with the varaible child which defined above.

Reply:
thx, done